File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main (void)
{
    string text = get_string("Text: ");
    int letter_count = 0;
    int word_count = 1;
    int sentence_count = 0;
    float index;



    for (int i = 0; i<strlen(text); i++)
    {
        if(tolower(text[i]) >= 97 && tolower(text[i]) <= 122 )
        {
            letter_count++;
        }

        if(text[i] == 32)
        {
            word_count++;
        }

        if(text[i] == 33 || text[i] == 63  || text[i] == 46 )
        {
           sentence_count++;
        }
    }

    float L = 0;
    float S = 0;

    L = 100 * (float)letter_count/(float)word_count;
    S = 100 * (float)sentence_count/(float)word_count;

    index = 0.0588 * L - 0.296 * S - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }

    if (index >= 1 && index < 16)
    {
        printf("Grade %.00f\n", index);
        return 0;
    }

    if (index > 16)
    {
        printf("Grade 16+\n");
        return 0;
    }

    printf("%i letter(s)\n", letter_count);
    printf("%i word(s)\n", word_count);
    printf("%i sentence(s)\n", sentence_count);

}

File: readability.py
text = input("Text: ")
letter_count = 0;
word_count = 1;
sentence_count = 0;
index = 0.0;

for elem in text:
    if ord(elem.lower()) >= 97 and ord(elem.lower()) <= 122:
        letter_count += 1

    if ord(elem) == 32:
        word_count += 1

    if ord(elem) == 33 or ord(elem) == 63 or ord(elem) == 46:
        sentence_count += 1

L = 0.00
S = 0.00
L = 100 * float(letter_count) / float(word_count)
S = 100 * float(sentence_count) / float(word_count)

index = 0.0588 * L - 0.296 * S - 15.8


if index < 1:
    print("Before Grade 1")

if (index >= 1 and index < 16):
    print(f"Grade {int(round(index))}")

if index > 16:
    print("Grade 16+")
